synapse/handlers/room.py

```diff
@@ -1171,8 +1190,25 @@ async def shutdown_room(
         if not await self.store.get_room(room_id):
             raise NotFoundError("Unknown room id %s" % (room_id,))
 
+        # Check if the user is shadow-banned before any mutations occur.
```

Comment:
given this can only be called by an admin user, I think it might be a bit redundant to check the shadow-ban.